src/Core/Domain/Product/Combination/QueryResult/CombinationForAssociation.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult;
+
+class CombinationForAssociation
+{
+    /**
+     * @var int
+     */
+    private $productId;
+
+    /**
+     * @var int
+     */
+    private $combinationId;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $reference;
+
+    /**
+     * @var string
+     */
+    private $imageUrl;
+
+    /**
+     * @param int $productId
+     * @param int $combinationId
+     * @param string $name
```

Comment:
The final query will also need the combination name built based on its attributes. But to do that I need elements from this WIP PR https://github.com/PrestaShop/PrestaShop/pull/25742 (like `CombinationNameBuilder`)

So the missing data will be added in a future PR (along with improved behat tests), but it doesn't prevent validating and merging this one right now.